src/html/classic/ng/src/web/pages/extras/trashactions.js

```diff
@@ -0,0 +1,181 @@
+  }) => {
+
+    const check_by_type = {
+      agent: function() {
+        return {restorable: true, deletable: false};
+      },
+      alert: function() {
+        const restorable =
+         is_defined(entity.filter) ? !entity.filter.isInTrash() : true;
+        return {restorable, deletable: entity.isInUse()};
+      },
+      config: function() {
+        const restorable =
+          is_defined(entity.scanner) ? !entity.scanner.isInTrash() : true;
+        return {restorable, deletable: entity.isInUse()};
+      },
+      credential: function() {
+        return {restorable: true, deletable: entity.isInUse()};
+      },
+      filter: function() {
+        return {restorable: true, deletable: entity.isInUse()};
+      },
+      group: function() {
+        return {restorable: true, deletable: entity.isInUse()};
+      },
+      note: function() {
+        return {restorable: true, deletable: false};
+      },
+      override: function() {
+        return {restorable: true, deletable: false};
+      },
+      permission: function() {
+        return {restorable: true, deletable: entity.isInUse()};
+      },
+      port_list: function() {
+        return {restorable: true, deletable: entity.isInUse()};
+      },
+      report_format: function() {
+        return {restorable: true, deletable: entity.isInUse()};
+      },
+      role: function() {
+        return {restorable: true, deletable: entity.isInUse()};
+      },
+      scanner: function() {
+        const restorable = is_defined(entity.credential) ?
+          !entity.credential.isInTrash() : true;
+        return {restorable, deletable: entity.isInUse()};
+      },
+      schedule: function() {
+        return {restorable: true, deletable: entity.isInUse()};
+      },
+      tag: function() {
+        return {restorable: true, deletable: entity.isInUse()};
+      },
+      target: function() {
+        const ssh_cred = is_defined(entity.ssh_credential) ?
+         !entity.ssh_credential.isInTrash() : true;
+        const smb_cred = is_defined(entity.smb_credential) ?
+         !entity.smb_credential.isInTrash() : true;
+        const esxi_cred = is_defined(entity.esxi_credential) ?
+         !entity.esxi_credential.isInTrash() : true;
+        const snmp_cred = is_defined(entity.snmp_credential) ?
+         !entity.snmp_credential.isInTrash() : true;
+        const portlist = is_defined(entity.port_list) ?
+         !entity.port_list.isInTrash() : true;
+
+        const restorable =
+          ssh_cred && smb_cred && esxi_cred && snmp_cred && portlist;
+        return {restorable, deletable: entity.isInUse()};
+      },
+      task: function() {
+        const restorable = !entity.schedule.isInTrash() &&
+          !entity.target.isInTrash() &&
+          !entity.config.isInTrash() &&
+          !entity.scanner.isInTrash() &&
+          !entity.alerts.some(alert => alert.isInTrash());
+        return {restorable, deletable: false};
+      },
+    };
+
+    function mode() {
```

Comment:
As above. The function is created for each render. So move it above the component and pass entity as a parameter.

Reply:
Moved it out of TrashActions. This made it necessary, though, to also pass onEntityDelete and onEntityRestore as parameters.